src/main/java/com/whiletrue/tododemo/service/impl/TaskServiceImpl.java

```diff
@@ -30,4 +35,67 @@ public TaskResponse createTask(TaskRequest taskRequest) {
         taskRepository.save(task);
         return new TaskResponse(task);
     }
+
+    @Override
+    public List<TaskResponse> getTasks() {
+        List<Task> tasks = taskRepository.findAll();
+        List<TaskResponse> taskResponses = new ArrayList<>();
+
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            TaskResponse taskResponse = new TaskResponse(task);
+            taskResponses.add(taskResponse);
+        }
+        return taskResponses;
+    }
+
+    @Override
+    public TaskResponse getTask(Long taskId) {
+
+        Optional<Task> task = taskRepository.findById(taskId);
```

Comment:
This is fine, but in the newer version of Spring you have a very handy ResponseStatusException, so you can do something like that:
```java
var task = taskRepository.findById(taskId)..orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Task not found"));
return new TaskResponse(task)
```
so you don't need a custom exception